robottelo/ssh.py

```diff
@@ -225,20 +265,53 @@ def upload_file(local_file, remote_file, key_filename=None, hostname=None):
         placed.
     :param hostname: target machine hostname. If not provided will be used the
         ``server.hostname`` from the configuration.
+    :param str key_filename: The path of the ssh private key to use when
+        connecting to the server. If it is ``None`` ``key_filename`` from
+        configuration's ``server`` section will be used.
     """
-    with get_connection(
-        hostname=hostname, key_filename=key_filename
-    ) as connection:  # pragma: no cover
-        try:
-            sftp = connection.open_sftp()
-            # Check if local_file is a file-like object and use the proper
-            # paramiko function to upload it to the remote machine.
-            if hasattr(local_file, 'read'):
-                sftp.putfo(local_file, remote_file)
-            else:
-                sftp.put(local_file, remote_file)
-        finally:
-            sftp.close()
+
+    with get_sftp_session(hostname, key_filename) as sftp:
+        _upload_file(sftp, local_file, remote_file)
+
+
+def upload_files(local_dir, remote_dir, file_search="*.txt",
+                 hostname=None, key_filename=None):
+    """ Upload all files from directory to a remote directory
```

Comment:
You should probably mention that it recursively searches for files but then squashes it all in a single directory. i.e. local `/tmp/first/file` and `/tmp/subdir/something` results in remote `/tmp/file` and `/tmp/something`.